olp-cpp-sdk-core/src/http/android/HttpClient.java

```diff
@@ -315,12 +319,38 @@ protected Void doInBackground(Request... requests) {
                 conn.setRequestProperty("Connection", "Close");
             }
 
+            Log.d(LOGTAG, "Printing Request Headers...\n");
+            Map<String,List<String>> requestHeaders = conn.getRequestProperties();
+            for (Map.Entry<String, List<String>> entry : requestHeaders.entrySet()) {
+              String header = entry.getKey();
+              List<String> values = entry.getValue();
+              Log.d(LOGTAG, header + ":" + values);
+              if(header != null) {
+                uploadedContentSize += header.length();
+                //check if Content-length header present
+                if (header.compareToIgnoreCase("Content-Length") == 0
+                    && values.get(0) != null) {
+                  uploadedContentSize += Integer.parseInt(values.get(0));
+                  uploadContentSizePresent = true;
+                }
```

Comment:
Does it make sense to do that?

Reply:
It is theoreticaly should be set by setFixedLengthStreamingMode. But it seams it is not available after. Se https://github.com/frohoff/jdk8u-dev-jdk/blob/master/src/share/classes/sun/net/www/protocol/http/HttpURLConnection.java restrictedHeaders. Maybe it could be changed with other HttpURLConnection implementation.